cou/utils/openstack.py

```diff
@@ -189,17 +199,21 @@ def __hash__(self) -> int:
         """
         return hash(f"{self.codename}{self.date}")
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: object) -> bool:
```

Comment:
it's for backward compatible for the code only (removing `str` implementation breaks a lot of unit tests), I think @gabrielcocenza has the reason why comparing for `str` is necessary

Reply:
`Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return) ` pylint does not like you idea

Reply:
then if I do `if-return+if-return+return` mypy don't like my idea: `error: Statement is unreachable  [unreachable]`

Reply:
And if I do ``
mypy also don't like it
```
lint: commands[2]> mypy --install-types --non-interactive .                                                                                  
cou/utils/openstack.py:202: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"  [override]
Found 2 errors in 1 file (checked 29 source files)                                                                                           
cou/utils/openstack.py:202: note: This violates the Liskov substitution principle                                                                                                                                                                                                         
cou/utils/openstack.py:202: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides                                                                                                                                                                     
cou/utils/openstack.py:202: note: It is recommended for "__eq__" to work with arbitrary objects, for example:                                                                                                                                                                             
cou/utils/openstack.py:202: note:     def __eq__(self, other: object) -> bool:                                                                                                                                                                                                            
cou/utils/openstack.py:202: note:         if not isinstance(other, OpenStackRelease):                                                                                                                                                                                                     
cou/utils/openstack.py:202: note:             return NotImplemented                                                                          
cou/utils/openstack.py:202: note:         return <logic to compare two OpenStackRelease instances>                                                                                                                                                                                        
cou/utils/openstack.py:205: error: Statement is unreachable  [unreachable]                                                                   
lint: exit 1 (0.49 seconds) /home/raychan/Work/canonical/forks/charmed-openstack-upgrader> mypy --install-types --non-interactive . pid=3159514                                                                                                                                           
  lint: FAIL code 1 (7.99=setup[0.02]+cmd[0.34,7.14,0.49] seconds)                                                                           
  evaluation failed :( (8.02 seconds)  
```

Reply:
> You'd need to update the if conditions. Sorry, I only quickly wrote pseudocode as an example.

Yes, I did

Reply:
This will work though

```python
    def __eq__(self, other: object) -> bool:                                                                                                                                                                                                                                              
        """Do equals."""
        if isinstance(other, OpenStackRelease):
            return self.index == other.index
        if isinstance(other, str):
            return self.index == OpenStackRelease(other).index
        return NotImplemented
```

Reply:
Long story short this module `openstack.py` is suppose to be a lib in the future. Not sure if ever will be, but in any case it gives a nice interface for the user be able to compare with strings.